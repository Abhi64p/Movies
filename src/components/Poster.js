import React from 'react';
import {View, Text, Image, StyleSheet, Dimensions} from 'react-native';

import {Images} from '../assets/images';
import {Typography} from '../styles';

const {width: deviceWidth} = Dimensions.get('window');
const width = (deviceWidth - 4 * 20) / 3;
const height = width * 1.49;

const Poster = ({name, posterUrl}) => (
  <View style={styles.posterContainer}>
    <Image
      source={Images?.[posterUrl] ?? Images.placeholder}
      style={styles.imageStyle}
      resizeMode="cover"
    />
    <Text style={[Typography.h3, styles.name]} numberOfLines={2}>
      {name}
    </Text>
  </View>
);

const styles = StyleSheet.create({
  posterContainer: {margin: 10},
  name: {width, minHeight: 40, marginTop: 3},
  imageStyle: {height, width},
});

export default Poster;
