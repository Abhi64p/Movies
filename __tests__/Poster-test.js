import React from 'react';
import {render} from '@testing-library/react-native';

import Poster from '../src/components/Poster';

test('Poster rendering correctly', () => {
  const {toJSON} = render(
    <Poster
      name="sample poster"
      posterUrl="http://posterUrl"
      width={100}
      height={150}
    />,
  );
  expect(toJSON()).toMatchSnapshot();
});
