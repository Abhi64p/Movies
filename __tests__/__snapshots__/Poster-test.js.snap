// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Poster rendering correctly 1`] = `
<View
  style={
    Object {
      "margin": 10,
    }
  }
>
  <Image
    resizeMode="cover"
    source={
      Object {
        "testUri": "../../../src/assets/images/placeholder_for_missing_posters.png",
      }
    }
    style={
      Object {
        "height": 150,
        "width": 100,
      }
    }
  />
  <Text
    numberOfLines={2}
    style={
      Array [
        Object {
          "color": "white",
          "fontFamily": "TitilliumWeb-Light",
          "fontSize": 14,
        },
        Object {
          "height": 40,
          "marginTop": 3,
        },
        Object {
          "width": 100,
        },
      ]
    }
  >
    sample poster
  </Text>
</View>
`;
